File: sdkwork-app-sdk-flutter/lib/src/api/voice_speakers.dart
import '../http/client.dart';
import '../models.dart';

class VoiceSpeakersApi {
  final HttpClient _client;
  
  VoiceSpeakersApi(this._client);

  /// 获取发音人详情
  Future<PlusApiResultVoiceSpeakerDetailVO?> getSpeakerDetail(String speakerId) async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers/${speakerId}'));
    return response is PlusApiResultVoiceSpeakerDetailVO ? response : null;
  }

  /// 更新发音人
  Future<PlusApiResultVoiceSpeakerVO?> updateSpeaker(String speakerId, VoiceSpeakerCreateForm body) async {
    final response = await _client.put(ApiPaths.appPath('/voice-speakers/${speakerId}'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoiceSpeakerVO ? response : null;
  }

  /// 删除发音人
  Future<PlusApiResultVoid?> deleteSpeaker(String speakerId) async {
    final response = await _client.delete(ApiPaths.appPath('/voice-speakers/${speakerId}'));
    return response is PlusApiResultVoid ? response : null;
  }

  /// 获取发音人列表
  Future<PlusApiResultPageVoiceSpeakerVO?> listSpeakers(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers'), params: params);
    return response is PlusApiResultPageVoiceSpeakerVO ? response : null;
  }

  /// 创建发音人
  Future<PlusApiResultVoiceSpeakerVO?> createSpeaker(VoiceSpeakerCreateForm body) async {
    final response = await _client.post(ApiPaths.appPath('/voice-speakers'), body: body, contentType: 'application/json');
    return response is PlusApiResultVoiceSpeakerVO ? response : null;
  }

  /// 更新发音人状态
  Future<PlusApiResultVoid?> updateStatus(String speakerId, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.appPath('/voice-speakers/${speakerId}/status'), params: params);
    return response is PlusApiResultVoid ? response : null;
  }

  /// 设为默认发音人
  Future<PlusApiResultVoiceSpeakerVO?> setAsDefault(String speakerId) async {
    final response = await _client.post(ApiPaths.appPath('/voice-speakers/${speakerId}/set-default'));
    return response is PlusApiResultVoiceSpeakerVO ? response : null;
  }

  /// 获取发音人统计
  Future<PlusApiResultSpeakerStatisticsVO?> getStatistics() async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers/statistics'));
    return response is PlusApiResultSpeakerStatisticsVO ? response : null;
  }

  /// 获取默认发音人
  Future<PlusApiResultVoiceSpeakerVO?> getDefaultSpeaker() async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers/default'));
    return response is PlusApiResultVoiceSpeakerVO ? response : null;
  }

  /// 根据代码获取发音人
  Future<PlusApiResultVoiceSpeakerVO?> getSpeakerByCode(String code) async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers/code/${code}'));
    return response is PlusApiResultVoiceSpeakerVO ? response : null;
  }

  /// 获取渠道发音人
  Future<PlusApiResultListVoiceSpeakerVO?> listSpeakersByChannel(String channel) async {
    final response = await _client.get(ApiPaths.appPath('/voice-speakers/channel/${channel}'));
    return response is PlusApiResultListVoiceSpeakerVO ? response : null;
  }
}

File: sdkwork-app-sdk-python/sdkwork_app_sdk/api/voice_speakers.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultListVoiceSpeakerVO, PlusApiResultPageVoiceSpeakerVO, PlusApiResultSpeakerStatisticsVO, PlusApiResultVoiceSpeakerDetailVO, PlusApiResultVoiceSpeakerVO, PlusApiResultVoid, VoiceSpeakerCreateForm

class VoiceSpeakersApi:
    """voice_speakers API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_speaker_detail(self, speakerId: str) -> PlusApiResultVoiceSpeakerDetailVO:
        """获取发音人详情"""
        return self._client.get(f"/app/v3/api/voice-speakers/{speakerId}")

    def update_speaker(self, speakerId: str, body: VoiceSpeakerCreateForm) -> PlusApiResultVoiceSpeakerVO:
        """更新发音人"""
        return self._client.put(f"/app/v3/api/voice-speakers/{speakerId}", json=body)

    def delete_speaker(self, speakerId: str) -> PlusApiResultVoid:
        """删除发音人"""
        return self._client.delete(f"/app/v3/api/voice-speakers/{speakerId}")

    def list_speakers(self, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPageVoiceSpeakerVO:
        """获取发音人列表"""
        return self._client.get(f"/app/v3/api/voice-speakers", params=params)

    def create_speaker(self, body: VoiceSpeakerCreateForm) -> PlusApiResultVoiceSpeakerVO:
        """创建发音人"""
        return self._client.post(f"/app/v3/api/voice-speakers", json=body)

    def update_status(self, speakerId: str, params: Optional[Dict[str, Any]] = None) -> PlusApiResultVoid:
        """更新发音人状态"""
        return self._client.post(f"/app/v3/api/voice-speakers/{speakerId}/status", params=params)

    def set_as_default(self, speakerId: str) -> PlusApiResultVoiceSpeakerVO:
        """设为默认发音人"""
        return self._client.post(f"/app/v3/api/voice-speakers/{speakerId}/set-default")

    def get_statistics(self) -> PlusApiResultSpeakerStatisticsVO:
        """获取发音人统计"""
        return self._client.get(f"/app/v3/api/voice-speakers/statistics")

    def get_default_speaker(self) -> PlusApiResultVoiceSpeakerVO:
        """获取默认发音人"""
        return self._client.get(f"/app/v3/api/voice-speakers/default")

    def get_speaker_by_code(self, code: str) -> PlusApiResultVoiceSpeakerVO:
        """根据代码获取发音人"""
        return self._client.get(f"/app/v3/api/voice-speakers/code/{code}")

    def list_speakers_by_channel(self, channel: str) -> PlusApiResultListVoiceSpeakerVO:
        """获取渠道发音人"""
        return self._client.get(f"/app/v3/api/voice-speakers/channel/{channel}")
